File: src/services/TheBrain/src/domain/occurrences/OccurrenceEventMessage.ts
import IncidentMessage from '@domain/messenger/IncidentMessage';

export default class OccurrenceEventMessage {
    public OccurrenceId: number = 0;
    public Latitude: number = 0;
    public Longitude: number = 0;
    public Location: string | null = null;
    public OccurrenceName: string | null = null;
    public StatusName: string | null = null;
    public OccurrenceUrgencyLevelName: string | null = null;
    public OccurrenceTypeName: string | null = null;
    public UserId: number = 0;
    public UserName: string | null = null;
    public Active: boolean = false;

    toIncidentMessage() : IncidentMessage {
        const messageString = JSON.stringify(this);
        return new IncidentMessage(messageString);
    }
}

File: src/services/NationalGeographicMessager/Domain/OcurrencesAggregated/OccurrenceEventMessage.cs
using NationalGeographicMessager.Domain.NotificationMessageAggregated;
using System.Text.Json;

namespace NationalGeographicMessager.Domain.OcurrencesAggregated
{
    public class OccurrenceEventMessage
    {
        public int OccurrenceId { get; set; }

        public double Latitude { get; set; }

        public double Longitude { get; set; }

        public string? Location { get; set; }

        public string? OccurrenceName { get; set; }

        public string? StatusName { get; set; }

        public string? OccurrenceUrgencyLevelName { get; set; }

        public string? OccurrenceTypeName { get; set; }

        public int UserId { get; set; }

        public string? UserName { get; set; }

        public bool Active { get; set; }


        public IncidentMessage ToIncidentMessage()
        {
            return new IncidentMessage(Latitude, Longitude, this);
        }
    }
}
